### PR TITLE
gtk4: Add sample/examples/application7

### PR DESCRIPTION
**Pull request by ToshioCP**
Add gtk4/sample/examples/application7

### CODE REVIEW

File: gtk4/sample/examples/application7/exampleapp.rb [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+          </item>
+        </section>
+      </menu>
+    </interface>
+    MENU
+    super(application: application)
+    builder = Gtk::Builder.new(string: menu_ui)
+    gears.menu_model = builder["menu"]
+    @settings = Gio::Settings.new("org.gtk.exampleapp")
+    @settings.bind("transition", stack, "transition-type", Gio::SettingsBindFlags::DEFAULT)
```
Comment (kou) [inline gtk4/sample/examples/application7/exampleapp.rb:215]: Can we use `:default` here?

```suggestion
    @settings.bind("transition", stack, "transition-type", :default)
```

File: gtk4/sample/examples/application7/exampleapp.rb [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+  private
+
+  def search_text_changed(search_entry)
+    text = search_entry.text
+    return if text.empty?
+
+    tab = stack.visible_child
+    view = tab.child
+    buffer = view.buffer
+    range = buffer.start_iter.forward_search(text, Gtk::TextSearchFlags::CASE_INSENSITIVE)
```
Comment (kou) [inline gtk4/sample/examples/application7/exampleapp.rb:247]:
```suggestion
    range = buffer.start_iter.forward_search(text, :case_insensitive)
```

**Comment by ToshioCP** [comment]
How about `Gio::SettingsBindFlags::DEFAULT`?
It is used in three places.

**Comment by kou** [comment]
If we can use `:default`, I want to use `:default` in all places.

**Comment by ToshioCP** [comment]
I've changed all Gio::SettingsBingFlags::DEFAULT to :default.

**Comment by kou** [comment]
Did `:default` work?

**Comment by ToshioCP** [comment]
Yes.
They worked at least in my environment (UBUNTU 22.10).

**Comment by kou** [comment]
Thanks!

**Pull request merged**